python/ray/dataframe/dataframe.py

```diff
@@ -2418,9 +2444,31 @@ def rdiv(self, other, axis='columns', level=None, fill_value=None):
     def reindex(self, labels=None, index=None, columns=None, axis=None,
                 method=None, copy=True, level=None, fill_value=np.nan,
                 limit=None, tolerance=None):
-        raise NotImplementedError(
-            "To contribute to Pandas on Ray, please visit "
-            "github.com/ray-project/ray.")
+        if not columns:
+            return DataFrame()
+        col_idx = [self.columns.get_loc(columns[i])
```

Comment:
This needs to handle cases where some of the passed in values are not in self.columns. Pandas fills this case with NaNs.